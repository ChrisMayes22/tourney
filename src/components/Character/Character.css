.flexContainer__col{
    display: flex;
    flex-direction: column;
    height: 85vh;
    align-items: center;
    justify-content: flex-end;
    margin: 10px;
    background-image: linear-gradient(to left, #f9c1ce, #cc8293, #f9c1ce);
}

.characterPortrait{
    height: auto;
    object-fit: contain;
    border-radius: 8px;
    box-shadow: 2px 2px 10px rgb(0, 0, 0);
    max-height: 60vh;
    max-width: 27.6vw;
    min-width: 20vw;
    background-image: radial-gradient(#b36174, rgb(243, 164, 182));
    margin: 5px;
    flex-basis: 0;
    flex-grow: 1;
}


