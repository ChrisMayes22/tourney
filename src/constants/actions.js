export const INITIALIZE_CHARACTER = 'INITIALIZE_CHARACTER';
export const REMOVE_CHARACTER = 'REMOVE_CHARACTER';
export const REMOVE_LOWEST_SCORE = 'REMOVE_LOWEST_SCORE';
export const RATE_CHARACTER = 'RATE_CHARACTER';
export const SUBMIT_CHARACTER_RATINGS = 'SUBMIT_CHARACTER_RATINGS';
export const SET_PLAYERS = 'SET_PLAYERS';
export const RESET_FOR_FINALS = 'RESET_FOR_FINALS';
export const CHOOSE_WINNER = 'CHOOSE_WINNER';

export const initializeCharacter = function(imageUrl, id){
    return {type: INITIALIZE_CHARACTER, 
        payload: {image: imageUrl, id}}
}
export const removeCharacter = function(id){
    console.log('Remove Character Dispatched')
    console.log(REMOVE_CHARACTER)
    return {type: REMOVE_CHARACTER, id}
}
export const removeLowestScore = function(){
    return {type: REMOVE_LOWEST_SCORE};
}
export const rateCharacter = function(points, rowId, character){
    return {type: RATE_CHARACTER, payload:{
        points, rowId, character}}
}
export const submitCharacterRatings = function(playersArray){
    return {type: SUBMIT_CHARACTER_RATINGS, payload:{
        players: playersArray
    }};
};
export const setPlayers = function(num){
    return {type: SET_PLAYERS, number: num}
};
export const resetForFinals = function(){
    return {type: RESET_FOR_FINALS}
}
export const chooseWinner = function(character){
    return {type: CHOOSE_WINNER, character}
}

