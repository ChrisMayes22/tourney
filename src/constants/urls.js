export const VOTING_PAGE = '/voting';
export const ELIMINATION_PAGE = '/elimination';
export const NOT_ENOUGH_CHARACTERS = '/not-enough-characters';
export const UPLOAD_PAGE = '/upload';
export const WINNER_PAGE = '/winner';
export const HOME = '/';