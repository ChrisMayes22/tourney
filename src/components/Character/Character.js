import React from 'react';
import classes from './Character.css';
import RatingRow from '../RatingRow/RatingRow';
import SubmitButton from '../SubmitButton/SubmitButton';
import * as urls from '../../constants/urls';
import { Link } from 'react-router-dom';

export const Character = (props) => {

    const arr = new Array(props.players).fill(null).map((_, index) => index);

    return(
            <section className={classes.flexContainer__col}>
                <img 
                    src={props.character.imageUrl} 
                    className={classes.characterPortrait}
                    alt="One of the images chosen for competition"
                />
                {props.finals ? <Link 
                                    to= {props.characters.length === 2 ? urls.WINNER_PAGE : urls.FINALS_PAGE}
                                    onClick={props.clicked}>
                                    <SubmitButton>
                                        Winner!
                                    </SubmitButton>
                                </Link> : arr.map((number) => {
                    return<RatingRow 
                            character={props.character}
                            rowId={`${props.character.id}-row-${number+1}`}
                            key={`key:${props.character.id}-row-${number+1}`}
                            id={`id:${props.character.id}-row-${number+1}`}
                            clicked={props.clicked}/>
                })}           
            </section> 
    );
}

export default Character;