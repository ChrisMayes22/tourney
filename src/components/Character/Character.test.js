import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

import React from 'react';
import { shallow } from 'enzyme';

import { Character } from './Character';
import * as urls from '../../constants/urls';

let props= null;
let wrapper = null;

describe('When Character first renders', function(){
    beforeEach(function(){
        props = {
            players: 3,
            character: {
                imageUrl: `foo$Image1`,
                roundRatings: {
                    rowOne: {rowId: `foo$id1-row-1`, points: 5},
                    rowTwo: {rowId: `foo$id1-row-2`, points: 5},
                    rowThree: {rowId: `foo$id1-row-3`, points: 5},
                    rowFour: {rowId: `foo$id1-row-4`, points: 5}
                },
                points: 20,
                hadTurn: false,
                isEliminated: {check: false, whenEliminated: null},
                id: 'foo$id1'
              },
            characters: [{
                imageUrl: `foo$Image1`,
                roundRatings: {
                    rowOne: {rowId: `foo$id1-row-1`, points: 5},
                    rowTwo: {rowId: `foo$id1-row-2`, points: 5},
                    rowThree: {rowId: `foo$id1-row-3`, points: 5},
                    rowFour: {rowId: `foo$id1-row-4`, points: 5}
                },
                points: 20,
                hadTurn: false,
                isEliminated: {check: false, whenEliminated: null},
                id: 'foo$id1'
              },
              {
                imageUrl: `foo$Image2`,
                roundRatings: {
                    rowOne: {rowId: `foo$id2-row-1`, points: 5},
                    rowTwo: {rowId: `foo$id2-row-2`, points: 5},
                    rowThree: {rowId: `foo$id2-row-3`, points: 5},
                    rowFour: {rowId: `foo$id2-row-4`, points: 5}
                },
                points: 20,
                hadTurn: false,
                isEliminated: {check: false, whenEliminated: null},
                id: 'foo$id2'
              }],
              finals: true,
              
        }

        wrapper = shallow(<Character {...props}/>)
    })

    test('Character renders as expected', function(){
        expect(wrapper).toMatchSnapshot();
    })
    describe('Link renders as expected', function(){
        test('Given that props.finals === true, exactly one Link renders', function(){
            expect(wrapper.find('Link')).toHaveLength(1);
        })
        test('Given that props.finals === false, no Links render', function(){
            props = {...props, finals: false}
            wrapper = shallow(<Character {...props}/>)
            expect(wrapper.find('Link')).toHaveLength(0);
        })
        test('Given a Link renders and props.characters.length === 2, to === urls.WINNER_PAGE', function(){
            expect(wrapper.exists(`[to="${urls.FINALS_PAGE}"]`)).toBe(false);
            expect(wrapper.exists(`[to="${urls.WINNER_PAGE}"]`)).toBe(true);
        })
        test('Given a Link renders and props.characters.length > 2, to === urls.FINALS_PAGE', function(){
            props={ ...props,
                characters: [{
                    imageUrl: `foo$Image1`,
                    roundRatings: {
                        rowOne: {rowId: `foo$id1-row-1`, points: 5},
                        rowTwo: {rowId: `foo$id1-row-2`, points: 5},
                        rowThree: {rowId: `foo$id1-row-3`, points: 5},
                        rowFour: {rowId: `foo$id1-row-4`, points: 5}
                    },
                    points: 20,
                    hadTurn: false,
                    isEliminated: {check: false, whenEliminated: null},
                    id: 'foo$id1'
                  },
                  {
                    imageUrl: `foo$Image2`,
                    roundRatings: {
                        rowOne: {rowId: `foo$id2-row-1`, points: 5},
                        rowTwo: {rowId: `foo$id2-row-2`, points: 5},
                        rowThree: {rowId: `foo$id2-row-3`, points: 5},
                        rowFour: {rowId: `foo$id2-row-4`, points: 5}
                    },
                    points: 20,
                    hadTurn: false,
                    isEliminated: {check: false, whenEliminated: null},
                    id: 'foo$id2'
                  }, 
                  {
                    imageUrl: `foo$Image3`,
                    roundRatings: {
                        rowOne: {rowId: `foo$id3-row-1`, points: 5},
                        rowTwo: {rowId: `foo$id3-row-2`, points: 5},
                        rowThree: {rowId: `foo$id3-row-3`, points: 5},
                        rowFour: {rowId: `foo$id3-row-4`, points: 5}
                    },
                    points: 20,
                    hadTurn: false,
                    isEliminated: {check: false, whenEliminated: null},
                    id: 'foo$id3'
                  }]
                }

                  wrapper = shallow(<Character {...props}/>)

            expect(wrapper.exists(`[to="${urls.FINALS_PAGE}"]`)).toBe(true);
            expect(wrapper.exists(`[to="${urls.WINNER_PAGE}"]`)).toBe(false);
        })
    })
    describe('RatingRows render as expected', function(){
        beforeEach(function(){
            props = {...props, finals: false}
            wrapper = shallow(<Character {...props}/>)
        })
        test('Given props.finals === true, RatingRows do not render', function(){
            props = {...props, finals: true}
            wrapper = shallow(<Character {...props}/>)
            
            expect(wrapper.exists('[id*="row"]')).toBe(false);
        })
        test('Given props.finals === false, RatingRows render', function(){
            expect(wrapper.exists('[id*="row"]')).toBe(true)
        })
        test('Given that RatingRows render and that props.players === 4, there should be 4 RatingRows rendered', function(){
            props = {...props, players: 4}
            wrapper = shallow(<Character {...props}/>)
            expect(wrapper.find('[id*="row"]')).toHaveLength(4)
        })
        test('Given that RatingRows render and that props.players === 3, there should be 4 RatingRows rendered', function(){
            props = {...props, players: 3}
            wrapper = shallow(<Character {...props}/>)
            expect(wrapper.find('[id*="row"]')).toHaveLength(3)
        })
        test('RatingRows ids render as expected', function(){
            props = {...props, players: 4}
            wrapper = shallow(<Character {...props}/>)

            expect(wrapper.exists(`[id="id:${props.character.id}-row-0"]`)).toBe(false)
            expect(wrapper.exists(`[id="id:${props.character.id}-row-1"]`)).toBe(true)
            expect(wrapper.exists(`[id="id:${props.character.id}-row-2"]`)).toBe(true)
            expect(wrapper.exists(`[id="id:${props.character.id}-row-3"]`)).toBe(true)
            expect(wrapper.exists(`[id="id:${props.character.id}-row-4"]`)).toBe(true)
            expect(wrapper.exists(`[id="id:${props.character.id}-row-5"]`)).toBe(false)

            wrapper.setProps({...props, players: 3})

            expect(wrapper.exists(`[id="id:${props.character.id}-row-0"]`)).toBe(false)
            expect(wrapper.exists(`[id="id:${props.character.id}-row-1"]`)).toBe(true)
            expect(wrapper.exists(`[id="id:${props.character.id}-row-2"]`)).toBe(true)
            expect(wrapper.exists(`[id="id:${props.character.id}-row-3"]`)).toBe(true)
            expect(wrapper.exists(`[id="id:${props.character.id}-row-4"]`)).toBe(false)
        })
        
    })
    
})