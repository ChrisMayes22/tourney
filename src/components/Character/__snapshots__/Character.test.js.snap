// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Character first renders Character renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Character
    character={
      Object {
        "hadTurn": false,
        "id": "foo$id1",
        "imageUrl": "foo$Image1",
        "isEliminated": Object {
          "check": false,
          "whenEliminated": null,
        },
        "points": 20,
        "roundRatings": Object {
          "rowFour": Object {
            "points": 5,
            "rowId": "foo$id1-row-4",
          },
          "rowOne": Object {
            "points": 5,
            "rowId": "foo$id1-row-1",
          },
          "rowThree": Object {
            "points": 5,
            "rowId": "foo$id1-row-3",
          },
          "rowTwo": Object {
            "points": 5,
            "rowId": "foo$id1-row-2",
          },
        },
      }
    }
    characters={
      Array [
        Object {
          "hadTurn": false,
          "id": "foo$id1",
          "imageUrl": "foo$Image1",
          "isEliminated": Object {
            "check": false,
            "whenEliminated": null,
          },
          "points": 20,
          "roundRatings": Object {
            "rowFour": Object {
              "points": 5,
              "rowId": "foo$id1-row-4",
            },
            "rowOne": Object {
              "points": 5,
              "rowId": "foo$id1-row-1",
            },
            "rowThree": Object {
              "points": 5,
              "rowId": "foo$id1-row-3",
            },
            "rowTwo": Object {
              "points": 5,
              "rowId": "foo$id1-row-2",
            },
          },
        },
        Object {
          "hadTurn": false,
          "id": "foo$id2",
          "imageUrl": "foo$Image2",
          "isEliminated": Object {
            "check": false,
            "whenEliminated": null,
          },
          "points": 20,
          "roundRatings": Object {
            "rowFour": Object {
              "points": 5,
              "rowId": "foo$id2-row-4",
            },
            "rowOne": Object {
              "points": 5,
              "rowId": "foo$id2-row-1",
            },
            "rowThree": Object {
              "points": 5,
              "rowId": "foo$id2-row-3",
            },
            "rowTwo": Object {
              "points": 5,
              "rowId": "foo$id2-row-2",
            },
          },
        },
      ]
    }
    finals={true}
    players={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="One of the images chosen for competition"
          src="foo$Image1"
        />,
        <Link
          replace={false}
          to="/winner"
        >
          <submitButton>
            Winner!
          </submitButton>
        </Link>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "One of the images chosen for competition",
          "className": undefined,
          "src": "foo$Image1",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <submitButton>
            Winner!
          </submitButton>,
          "onClick": undefined,
          "replace": false,
          "to": "/winner",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Winner!",
          },
          "ref": null,
          "rendered": "Winner!",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="One of the images chosen for competition"
            src="foo$Image1"
          />,
          <Link
            replace={false}
            to="/winner"
          >
            <submitButton>
              Winner!
            </submitButton>
          </Link>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "One of the images chosen for competition",
            "className": undefined,
            "src": "foo$Image1",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <submitButton>
              Winner!
            </submitButton>,
            "onClick": undefined,
            "replace": false,
            "to": "/winner",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Winner!",
            },
            "ref": null,
            "rendered": "Winner!",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
